File: packages/sdk-py/pylva/wrappers/_patch.py
"""Auto-patch coordinator — runs on ``import pylva``. Every try_patch_* is
wrapped in try/except so a missing peer dep is a silent no-op (R1)."""

from __future__ import annotations

_patched_once = False


def apply_all_patches() -> None:
    """Call once on import and defensively from init() for Jupyter / importlib.reload."""
    global _patched_once
    from .anthropic_wrapper import try_patch_anthropic
    from .openai_wrapper import try_patch_openai

    try:
        try_patch_openai()
    except Exception as err:
        print(f"[pylva] openai auto-patch failed: {err}", flush=True)

    try:
        try_patch_anthropic()
    except Exception as err:
        print(f"[pylva] anthropic auto-patch failed: {err}", flush=True)

    _patched_once = True


def was_auto_patched() -> bool:
    return _patched_once

File: packages/sdk-ts/src/wrappers/_patch.ts
// Auto-patch coordinator (D8 + D21).
// Called on `import 'pylva'` and defensively re-called from init() for
// HMR / import-order edge cases. Every patch attempt is try/catch — if the
// provider SDK isn't installed, patch is a silent no-op (R1 isolation).

import { applyOpenAiPatch } from './openai.js';
import { applyAnthropicPatch } from './anthropic.js';
import { applyVercelAiPatch } from './vercel-ai.js';

let patchedOnce = false;

export function applyAllPatches(): void {
  tryPatch(applyOpenAiPatch, 'openai');
  tryPatch(applyAnthropicPatch, 'anthropic');
  tryPatch(applyVercelAiPatch, 'vercel-ai');
  patchedOnce = true;
}

export function wasAutoPatched(): boolean {
  return patchedOnce;
}

function tryPatch(fn: () => void, label: string): void {
  try {
    fn();
  } catch (err) {
    console.warn(
      `[pylva] ${label} auto-patch failed: ${err instanceof Error ? err.message : String(err)}`,
    );
  }
}
